consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BlockTimer.java

```diff
@@ -51,6 +57,8 @@ public BlockTimer(
     this.bftExecutors = bftExecutors;
     this.currentTimerTask = Optional.empty();
     this.clock = clock;
+    this.blockPeriodSeconds = forksSchedule.getFork(0).getValue().getBlockPeriodSeconds();
+    this.emptyBlockPeriodSeconds = forksSchedule.getFork(0).getValue().getEmptyBlockPeriodSeconds();
```

Comment:
Hi, transitions are working fine AFAICS. I added a comment in the [Issue](https://github.com/hyperledger/besu/issues/3810#issuecomment-2053663437) with some logs before and when the transition happened. Please take a look.

Cheers.

Reply:
> Hi, transitions are working fine AFAICS. I added a comment in the [Issue](https://github.com/hyperledger/besu/issues/3810#issuecomment-2053663437) with some logs before and when the transition happened. Please take a look.
> 
> Cheers.

Hi, transitions are working fine AFAICS. I added a comment in the https://github.com/hyperledger/besu/issues/3810#issuecomment-2053663437 with some logs before and when the transition happened. Please take a look.

Cheers.

Reply:
> Not got my head around how these are being used yet, but this may need to account for "transitions", see https://besu.hyperledger.org/development/private-networks/how-to/configure/consensus/qbft#transitions
> 
> e.g. with a transition at block 1 and a chain height of 3, if we start besu and construct this class then we'll get the wrong initial blockPeriodSeconds...not sure if that matters yet.
> 
> If these initial values are never used, then maybe setting to -1 will make that more obvious.

Hi, transitions are working fine AFAICS. I added a comment in the https://github.com/hyperledger/besu/issues/3810#issuecomment-2053663437 with some logs before and when the transition happened. Please take a look.

The default value of emptyBlockPeriodSeconds  is 0, so in practice it will use the value of blockPeriodSeconds, as you can see in the link above.

I hope I undertood you question correctly :)

Cheers.